lib/matplotlib/axes/_axes.py

```diff
@@ -3895,88 +3885,66 @@ def bxp(self, bxpstats, positions=None, widths=None, vert=True,
 
         zdelta = 0.1
         # box properties
+        final_boxprops = dict(
+            linestyle=rcParams['boxplot.boxprops.linestyle'],
+            linewidth=rcParams['boxplot.boxprops.linewidth'],
+        )
         if patch_artist:
-            final_boxprops = dict(
-                linestyle=rcParams['boxplot.boxprops.linestyle'],
+            final_boxprops.update(
                 edgecolor=rcParams['boxplot.boxprops.color'],
-                facecolor=rcParams['patch.facecolor'],
-                linewidth=rcParams['boxplot.boxprops.linewidth']
+                facecolor=('white' if rcParams['_internal.classic_mode'] else
+                           rcParams['patch.facecolor']),
             )
-            if rcParams['_internal.classic_mode']:
-                final_boxprops['facecolor'] = 'white'
         else:
-            final_boxprops = dict(
-                linestyle=rcParams['boxplot.boxprops.linestyle'],
+            final_boxprops.update(
                 color=rcParams['boxplot.boxprops.color'],
             )
 
         final_boxprops['zorder'] = zorder
         if boxprops is not None:
             final_boxprops.update(boxprops)
 
-        # other (cap, whisker) properties
-        final_whiskerprops = dict(
-            linestyle=rcParams['boxplot.whiskerprops.linestyle'],
-            linewidth=rcParams['boxplot.whiskerprops.linewidth'],
-            color=rcParams['boxplot.whiskerprops.color'],
-        )
-
-        final_capprops = dict(
-            linestyle=rcParams['boxplot.capprops.linestyle'],
-            linewidth=rcParams['boxplot.capprops.linewidth'],
-            color=rcParams['boxplot.capprops.color'],
-        )
-
-        final_capprops['zorder'] = zorder
-        if capprops is not None:
-            final_capprops.update(capprops)
-
+        # whisker properties
+        final_whiskerprops = {
+            k.split('.')[-1]: v for k, v in rcParams.items()
+            if k.startswith('boxplot.whiskerprops')
+        }
         final_whiskerprops['zorder'] = zorder
         if whiskerprops is not None:
             final_whiskerprops.update(whiskerprops)
-
-        # set up the default flier properties
-        final_flierprops = dict(
-            linestyle=rcParams['boxplot.flierprops.linestyle'],
-            linewidth=rcParams['boxplot.flierprops.linewidth'],
-            color=rcParams['boxplot.flierprops.color'],
-            marker=rcParams['boxplot.flierprops.marker'],
-            markerfacecolor=rcParams['boxplot.flierprops.markerfacecolor'],
-            markeredgecolor=rcParams['boxplot.flierprops.markeredgecolor'],
-            markeredgewidth=rcParams['boxplot.flierprops.markeredgewidth'],
-            markersize=rcParams['boxplot.flierprops.markersize'],
-        )
-
+        # cap properties
```

Comment:
For me, it takes enough mental bandwidth to read these dict-comprehensions that I wonder if factoring out into a sub function makes sense

Reply:
pushed a second commit that does it, indeed that's a good idea.

Reply:
ahhh it looks great.